src/librustdoc/html/render.rs

```diff
@@ -1552,7 +1585,7 @@ impl Context {
                 if let Ok(mut redirect_out) = OpenOptions::new().create_new(true)
```

Comment:
An initial version of this put the buffer inside `layout::render` and `layout::redirect`, before realizing that most of the existing calls used their own ad-hoc buffering anyway. So that initial commit was scrapped and turned into the global buffer that is up now. After having gone through all the `layout::` calls in here, i don't think i'd mind just reverting the global buffer and adding `BufWriter`s here, since that was the major culprit being addressed in this commit. This was one of the main focuses in this PR, since the raw `write!()` on a redirect will translate to about 9 file writes for a fairly small file, due to how they get broken up in the formatting.